doctr/models/recognition/parseq/pytorch.py

```diff
@@ -266,7 +266,8 @@ def decode_autoregressive(self, features: torch.Tensor, max_len: Optional[int] =
         ).int()
 
         pos_logits = []
-        for i in range(max_length):
+        i = 0
+        while i < max_length:
```

Comment:
I remember there was a issue with `while` loops by exporting to onnx so we have to be careful here (needs to be checked)

Reply:
I changed it because it was some unecessary complication related to breaks in torch.compile. Changing to a while loop and changing the logic a bit helped. Hopefully it works for the onnx also